### PR TITLE
install-deps.sh: retry if dpkg was interrupted

### PR DESCRIPTION
**Pull request by tchaikov**
there is chance that apt-get is interrupted in the middle when a new PR
cancels the running jenkins job, the next job running apt-get or dpkg
would run into issues like:

E: dpkg was interrupted, you must manually run 'sudo dpkg --configure -a' to correct the problem.
Build step 'Execute shell' marked build as failure

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by rzarzynski — APPROVED** [review]

**Comment by tchaikov** [comment]
see also https://jenkins.ceph.com/job/ceph-pull-requests-arm64/15786/

**Comment by tchaikov** [comment]
@rzarzynski  could you take another look? i just realized that i should have add this check in `run-make.sh` which runs `apt-get` before `install-deps.sh` does.

File: src/script/run-make.sh [hunk 1]
```diff
@@ -54,6 +54,30 @@ function detect_ceph_dev_pkgs() {
     echo "$cmake_opts"
 }
 
+function do_install() {
```
Comment (rzarzynski) [inline src/script/run-make.sh:57]: We still have `install-deps.sh` in the commit's title.

**Review by rzarzynski — COMMENTED** [review]
To the record: I assume Jenkins prevents two (or more) jobs running on the same env. the same time. If so, then cleaning for an interrupted package manager run at first `apt-get` invocation should do the job.

File: src/script/run-make.sh (hunk 1 above)
Comment (tchaikov) [reply src/script/run-make.sh:57]: fixed and repushed.

**Review by rzarzynski — APPROVED** [review]

**Pull request merged**